arch/lkl/kernel/threads.c

```diff
@@ -130,6 +132,17 @@ void free_thread_stack(struct task_struct *tsk)
 		test_tsk_thread_flag(tsk, TIF_SIGPENDING), ti, current->comm);
 
 	kill_thread(ti);
+	
+	/* free any linked list of pending signals */
+	/* note that we do not need to take the lock as the thread is dead and ought to be unreachable*/
```

Comment:
I'm wondering. if this assumption is untrue. if there's a bug somewhere such that the thread is somehow reachable, what is the error going to look like if we don't take the lock and something goes boom? how easy would that be to find? would taking the lock (although technically not needed) result in an easier to find bug should said bug ever come into existence?

there's no change needed from these musings, but I think it is worth thinking about.

Reply:
Interesting point. I will think about that. Maybe I completely need to revisit the approach, reinstate do_signal that calls move then send and have the places that need to unlock explicitly do so. Then every one has to run holding the lock. 

David makes some points too about trap.c

Reply:
Done